tools/ui-components/src/button.test.js

```diff
@@ -0,0 +1,29 @@
+/* global expect */
+/* global jest */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { Button } from './button';
+
+describe('Button', () => {
+  it("should have the role 'button' and the correct text", () => {
+    render(<Button label='Hello world' />);
+
+    expect(
+      screen.getByRole('button', { name: /hello world/i })
+    ).toBeInTheDocument();
+  });
```

Comment:
This isn't in the scope of the PR, but I'm doing a walkthrough anyway, so that you know where I'm coming from with my testing approach.

RTL recommends choosing queries in [this order](https://testing-library.com/docs/queries/about#priority). `getByRole` is at the top, which means it should be the first go-to query.

In this test, `getByRole` looks through the DOM tree and finds an element with the ARIA role `button` and the text `Hello world`. The role search is important, as it helps prevent cases where a non-button element (div, span, p,...) is used in place of a real button but is not given appropriate ARIA attributes, resulting screen readers not being able to announce the element correctly.

Here are a few implementation examples that will pass this test

```html
<button>Hello world</button>

<div role="button">Hello world</div>

<span role="button">Hello world</span>

```